### PR TITLE
Several bug fixes and enhancement for checking user access before writing value

### PR DESCRIPTION
**Pull request by tander67**
Several bug fixes and enhancement for checking user access before writing value received from remote clients

### CODE REVIEW

**Comment by oroulet** [comment]
thanks for the patch. I will have a lool when I get 2 minutes

**Comment by oroulet** [comment]
Hi,
I just had a look at the pull request.
I think we need to do the write value check a bit differently.

All calls go through a session object. this is done to be able to add user
management later. I propose to add a constructor parameter to Session
called for example access_rigths.
in InternalServer an internal session is created, create the session using
for example access_right="admin" (or better use an enum)
then then pass access_rigths to set_attribute_value and in method do a test:

def set_attribute_value(self, nodeid, attr, value, access_right = "user")
...
if access_right != "admin":
     check_access
...


In addition you cannot modify the file called uaprotocol_auto.py. It is
autogenerate . Modify the file called uaprotocol_hand.py and look for the
class. It overrides the constructor. There you can override these values

Thanks
Olivier

**Comment by tander67** [comment]
Good morning,

As for adding access rights that sounds like the best solution.

I will look into modifying the uaprotocol_hand.py file for the other change
I made.

Thank you for the feedback.

From:  ORD <notifications@github.com>
Reply-To:  FreeOpcUa/python-opcua
<reply+0011d422282b168b5fe8ef32c9f78e9e29694d52c0bdc32392cf000000011150b78f9
2a169ce042f9fe8@reply.github.com>
Date:  Thursday, April 23, 2015 at 9:37 AM
To:  FreeOpcUa/python-opcua <python-opcua@noreply.github.com>
Cc:  Troy Anderson <tander67@mac.com>
Subject:  Re: [python-opcua] Several bug fixes and enhancement for checking
user access before writing value (#2)

Hi,
I just had a look at the pull request.
I think we need to do the write value check a bit differently.

All calls go through a session object. this is done to be able to add user
management later. I propose to add a constructor parameter to Session
called for example access_rigths.
in InternalServer an internal session is created, create the session using
for example access_right="admin" (or better use an enum)
then then pass access_rigths to set_attribute_value and in method do a test:

def set_attribute_value(self, nodeid, attr, value, access_right = "user")
...
if access_right != "admin":
 check_access
...


In addition you cannot modify the file called uaprotocol_auto.py. It is
autogenerate . Modify the file called uaprotocol_hand.py and look for the
class. It overrides the constructor. There you can override these values

Thanks
Olivier

On Thu, 23 Apr 2015 at 10:43 Olivier Roulet-Dubonnet <
olivier.roulet@gmail.com> wrote:

> thanks for the patch. I will have a lool when I get 2 minutes
>
> On 22 April 2015 at 22:43, tander67 <notifications@github.com> wrote:
>
>> Several bug fixes and enhancement for checking user access before writing
>> value received from remote clients
>> ------------------------------
>> You can view, comment on, or merge this pull request online at:
>>
>>   https://github.com/FreeOpcUa/python-opcua/pull/2
>> Commit Summary
>>
>>    - Added boolean support and fixed date time conversion
>>    - Changed default access level
>>    - Fixed adding methods
>>    - Added write access check
>>
>> File Changes
>>
>>    - *M* opcua/address_space.py
>>    <https://github.com/FreeOpcUa/python-opcua/pull/2/files#diff-0> (34)
>>    - *M* opcua/internal_server.py
>>    <https://github.com/FreeOpcUa/python-opcua/pull/2/files#diff-1> (3)
>>    - *M* opcua/node.py
>>    <https://github.com/FreeOpcUa/python-opcua/pull/2/files#diff-2> (2)
>>    - *M* opcua/uaprocessor.py
>>    <https://github.com/FreeOpcUa/python-opcua/pull/2/files#diff-3> (3)
>>    - *M* opcua/uaprotocol_auto.py
>>    <https://github.com/FreeOpcUa/python-opcua/pull/2/files#diff-4> (5)
>>    - *M* opcua/uatypes.py
>>    <https://github.com/FreeOpcUa/python-opcua/pull/2/files#diff-5> (10)
>>
>> Patch Links:
>>
>>    - https://github.com/FreeOpcUa/python-opcua/pull/2.patch
>>    - https://github.com/FreeOpcUa/python-opcua/pull/2.diff
>>
>> 
>> Reply to this email directly or view it on GitHub
>> <https://github.com/FreeOpcUa/python-opcua/pull/2>.
>>
>
>


Reply to this email directly or view it on GitHub
<https://github.com/FreeOpcUa/python-opcua/pull/2#issuecomment-95587608> .

**Comment by oroulet** [comment]
Hi,
I merged the 2 commits that do not break the build.

File: opcua/uatypes.py [hunk 1]
```diff
@@ -611,6 +616,9 @@ def _guess_type(self, val):
             return VariantType.ByteString
         elif type(val) == datetime:
             return VariantType.DateTime
+        elif type(val) == bool:
+            # TDA, added this because it was missing and causes exceptions when 'bool' type is used
```
Comment (oroulet) [inline opcua/uatypes.py]: please do not write that kind of comments. only write comments when there is osmething which is difficutl to understand by reading the code

**Comment by oroulet** [comment]
what could be merged , has been merged

**Pull request closed without merging**